### ISSUE TITLE
shorthand option for `max-statements-per-line`

### ISSUE THREAD

**Issue by rasenplanscher** [eslint/eslint#8092]
**What rule do you want to change?**
`max-statements-per-line`

**Does this change cause the rule to produce more or fewer warnings?**
neither

**How will the change be implemented? (New option, new default behavior, etc.)?**
Option short-hand, consistent with `max-depth` rule, as well as several other rules having shorthands for their most used options.

**Please provide some example code that this change will affect:**
Setting in `.eslintrc` etc.:
```yaml
max-statements-per-line: [ "error", 1 ]
```

**What does the rule currently do for this code?**
`eslint` throws an error because `1` is not an `Object`

**What will the rule do after it's changed?**
That setting should be equivalent to this:
```yaml
max-statements-per-line: [ "error", { "max": 1 } ]
```

**Comment by not-an-aardvark** [comment]
Thanks for the proposal. This seems like a reasonable idea to me. :+1:

**Comment by platinumazure** [comment]
I'm not strongly opposed, but we've been moving away from single numeric option values in order to ensure configuration is as explicit as possible. So I'm not really in support of moving the other direction here, but I won't oppose it if the rest of the team is :+1: to this.

**Comment by mysticatea** [comment]
We avoided magic numbers in options: #4182

**Comment by ilyavolodin** [comment]
We try to avoid magic numbers. We also try to avoid having multiple ways to do the same thing. I don't think that saving 5 keystrokes is worth having another way to configure this rule. Also, this might prevent us from adding more options later, due to the schema complexity.

**Comment by alberto** [comment]
Thanks for your interest in improving eslint. Unfortunately, it looks like this issue didn't get enough support from the team and so I'm closing it.
